NetKet/Sampler/abstract_sampler.hpp

```diff
@@ -50,8 +51,35 @@ class AbstractSampler {
     this->Reset(true);
   }
 
+  virtual void SetMachineFunc(
+      std::function<double(const Complex&)> machine_func) {
+    machine_func_.Set(machine_func);
+  }
+
+  virtual std::function<double(const Complex&)> GetMachineFunc() {
+    return machine_func_.Get();
+  }
+
+  class MachineFunc {
```

Comment:
What is the purpose of this wrapper class? `using MachineFunc = std::function<double(Complex)>;` should do the trick.

Reply:
It's mostly because I wanted machine_func_ to be initialized by default with std::norm, so I created a wrapper class with such a default initialization (instead of a null function default initialization)

Reply:
Default constructor of `AbstractSampler` can do that.

Reply:
good point